### PR TITLE
Proper logging

### PR DESCRIPTION
**Pull request by Yatekii**
This PR enables logging even with progressbars enabled!

### CODE REVIEW

File: src/logging.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if let Some(pb) = &*guard {
+        pb.println(message.as_ref());
+    } else {
+        eprintln!("{}", message.as_ref());
+    }
+}
+
+/// Writes an error to the log.
+/// This can be used for unwraps/eprintlns/etc.
+pub fn write_message(message: impl AsRef<str>) {
```
Comment (Tiwalun) [inline src/logging.rs]: I think naming this print would be more consistent.

File: src/logging.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    let mut guard = PROGRESS_BAR.write().unwrap();
+    *guard = Some(progress);
+}
+
+/// Writes an error to the log.
+/// This can be used for unwraps/eprintlns/etc.
+pub fn write_error(message: impl AsRef<str>) {
+    let guard = PROGRESS_BAR.write().unwrap();
+    if let Some(pb) = &*guard {
+        pb.println(message.as_ref());
```
Comment (Tiwalun) [inline src/logging.rs]: The error is now printed the same as the normal message. Shouldn't there be a difference?

File: src/logging.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+use log::{Level, LevelFilter};
+use std::{
+    fmt,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+static MAX_MODULE_WIDTH: AtomicUsize = AtomicUsize::new(0);
```
Comment (Tiwalun) [inline src/logging.rs]: What is the purpose of this variable?

For me it is not clear what module means here.

File: src/logging.rs (hunk 2 above)
Comment (Yatekii) [reply src/logging.rs]: Well this is currently not possible within indicatif, sadly.

File: src/logging.rs (hunk 3 above)
Comment (Yatekii) [reply src/logging.rs]: I renamed this to use a better name. Basically it is remembering how wide the window is to try and adjust the padding of all the logs.
I took this functionality from pretty_env_logger :)

**Pull request merged**